_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,346 @@
+echo ""
+----
+
+=== OpenShift Web Console
+
+To make sure your Keycloak server has been provisioned using the OpenShift web console,
+navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app
+to check its status. Once it is running, you can click on *Open URL* and then access Keycloak's *Administration Console*.
+
+== Configure Keycloak
+
+. Log into the Keycloak Admin Console using the username and password you specified earlier.
+
+. Create a new realm called *myrealm*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-openshift#_create_a_realm[create a realm].
+
+. Add a role called *user*. This role will be required to access our simple web application. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#assigning-permissions-using-roles-and-groups[create a role].
+
+. Add a new user named *alice*. Set an *email* address for this new user, we'll use *alice@example.org*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-openshift#_create_a_user[create a user].
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *openid-connect*
+** *Client ID*: *simple-webapp-oidc*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Action*, *Download adapter config*, and select `Keycloak OIDC JSON` as the *Format Option* to see the connection parameters.
+
+== Update the example application and push it to your fork
+
+. Navigate to the `src/main/webapp/WEB-INF` directory in the application.
+
+. Update the values in the `oidc.json` file as follows:
++
+[source,json]
+----
+{
+    "client-id" : "myclient",                                    <1>
+    "provider-url" : "${env.OIDC_PROVIDER_URL:http://localhost:8080}/realms/myrealm", <2>
+    "public-client" : "true",
+    "principal-attribute" : "preferred_username",
+    "ssl-required" : "EXTERNAL"
+}
+----
+<1> Change to the value of `client-id` defined in Keycloak: `simple-webapp-oidc`.
+<2> Change to the value of `provider-url` to the Keycloak realm URL. Use this command to obtain the URL:
```

Comment:
Instead of needing to update this line, I think we could also specify this using the OIDC_PROVIDER_URL environment variable in the Helm Chart configuration. 

We could potentially also update the elytron example to use "simple-webapp-oidc" for the client name instead of "myclient".

With those changes, we wouldn't actually need the user to make any updates to the oidc.json file.

WDYT?

Reply:
I changed the client name in the blog to myclient, just in case the example is used elsewhere.

Used environment variable to specify keycloak URL. Therefore, i've removed the entire section of configuring the application.